hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/StripeReader.java

```diff
@@ -432,7 +432,9 @@ void decodeAndFillBuffer(boolean fillBuffer) throws IOException {
 
     long start = Time.monotonicNow();
     // Step 2: decode into prepared output buffers
-    decoder.decode(decodeInputs, decodeIndices, outputs);
+    synchronized (decoder) {
```

Comment:
IMO, we should make RawErasureDecoder.decode() synchronized instead.

Reply:
Thanks for your review. @jojochuang 
Yes, that's better. I will update it in a few minutes.